src/devtools/client/inspector/rules/models/element-style.ts

```diff
@@ -153,12 +153,16 @@ export default class ElementStyle {
 
         for (const { rule, pseudoElement } of parentApplied) {
           if (!pseudoElement) {
-            this._maybeAddRule(rule, elemNodeWithId);
+            this._maybeAddRule(rule, parentNodeWithId);
           }
         }
       }
 
-      parentNodeId = elem.parentNode;
+      if (parentObject.preview?.node?.nodeName === "HTML") {
+        break;
+      }
```

Comment:
This is the fix, the rest is just renaming things. 

At first, I started playing with comparing this against data received from:
```ts
const ownerDocumentRequest = this.replayClient.getDocument(this.pauseId).catch(() => null);
```
but then I realized I could access this more directly.